src/main/java/nextstep/qna/domain/Answers.java

```diff
@@ -0,0 +1,54 @@
+package nextstep.qna.domain;
+
+import nextstep.qna.CannotDeleteException;
+import nextstep.users.domain.NsUser;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Answers {
+    private List<Answer> answers;
+
+    public Answers(List<Answer> answers) {
+        this.answers = answers;
+    }
+
+    public void checkDeletePermission(NsUser loginUser) throws CannotDeleteException {
+        boolean isNoDeletePermission = answers.stream()
+                .anyMatch(answer -> answer.isNotOwner(loginUser));
+
+        if(isNoDeletePermission){
+            throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
+        }
+    }
+
+    public void add(Answer answer) {
+        answers.add(answer);
+    }
+
+    public List<DeleteHistory> delete() {
+        markAllDeleted();
+        return createDeleteHistory();
+    }
+
+    private List<DeleteHistory> createDeleteHistory() {
+        List<DeleteHistory> histories = new ArrayList<>();
+        answers.forEach(answer -> histories.add(
+                new DeleteHistory(
+                        ContentType.ANSWER,
+                        answer.getId(),
+                        answer.getWriter(),
+                        LocalDateTime.now())
+        ));
```

Comment:
`Question`의 `delete()`처럼, `Answer`에도 `delete()` 메서드를 추가해보면 어떨까요?
```suggestion
        answers.forEach(answer -> histories.add(answer.delete(loginUser)));
```

Reply:
답변 1개도 유효성 체크 및 삭제가 필요하다 생각해서 추가하고
Answers 일급 컬렉션은 루프 돌면서 answer.delete 추가하는 형태로 변경했습니다